torchgeo/datamodules/geo.py

```diff
@@ -0,0 +1,540 @@
+
+        Returns:
+            A collection of data loaders specifying prediction samples.
+
+        Raises:
+            MisconfigurationException: If :meth:`setup` does not define a
+                'predict_dataset'.
+        """
+        dataset = self.predict_dataset or self.dataset
+        sampler = self.predict_sampler or self.sampler
+        batch_sampler = self.predict_batch_sampler or self.batch_sampler
+        if dataset is not None and (sampler or batch_sampler) is not None:
+            batch_size = self.predict_batch_size or self.batch_size
+            if batch_sampler is not None:
+                batch_size = 1
+                sampler = None
+            return DataLoader(
+                dataset=dataset,
+                batch_size=batch_size,
+                sampler=sampler,
+                batch_sampler=batch_sampler,
+                num_workers=self.num_workers,
+                collate_fn=stack_samples,
+            )
+        else:
+            msg = f"{self.__class__.__name__}.setup does not define a 'predict_dataset'"
+            raise MisconfigurationException(msg)
+
+    def transfer_batch_to_device(
+        self, batch: Dict[str, Tensor], device: torch.device, dataloader_idx: int
+    ) -> Dict[str, Tensor]:
+        """Transfer batch to device.
+
+        Defines how custom data types are moved to the target device.
+
+        Args:
+            batch: A batch of data that needs to be transferred to a new device.
+            device: The target device as defined in PyTorch.
+            dataloader_idx: The index of the dataloader to which the batch belongs.
+
+        Returns:
+            A reference to the data on the new device.
+        """
+        # Non-Tensor values cannot be moved to a device
+        del batch["crs"]
+        del batch["bbox"]
+
+        batch = super().transfer_batch_to_device(batch, device, dataloader_idx)
+        return batch
+
+    def on_after_batch_transfer(
+        self, batch: Dict[str, Tensor], dataloader_idx: int
+    ) -> Dict[str, Tensor]:
+        """Apply batch augmentations to the batch after it is transferred to the device.
+
+        Args:
+            batch: A batch of data that needs to be altered or augmented.
+            dataloader_idx: The index of the dataloader to which the batch belongs.
+
+        Returns:
+            A batch of data.
+        """
+        if self.trainer:
+            if self.trainer.training:
+                aug = self.train_aug or self.aug
+            elif self.trainer.validating or self.trainer.sanity_checking:
+                aug = self.val_aug or self.aug
+            elif self.trainer.testing:
+                aug = self.test_aug or self.aug
+            elif self.trainer.predicting:
+                aug = self.predict_aug or self.aug
+
+            batch = aug(batch)
+
+        return batch
+
+    def plot(self, *args: Any, **kwargs: Any) -> plt.Figure:
+        """Run the plot method of the dataset if one exists.
```

Comment:
Maybe we should comment that this method really only expects to be called from the trainer (if you are in "test" phase then nothing will happen silently because the self.dataset / self.val_dataset are not defined)